app/controllers/api/base_controller/renderer.rb

```diff
@@ -205,7 +217,7 @@ def expand_virtual_attributes(json, type, resource)
       def fetch_direct_virtual_attribute(type, resource, attr)
         return unless attr_accessible?(resource, attr)
         virtattr_accessor = virtual_attribute_accessor(type, attr)
```

Comment:
used by API when it needs extra logic for fetching an attribute, i.e. look at any of the fetch_<type>_attr named methods. currently implemented for service_dialogs content and orchestration_stacks stdout, so not traversing relationships and we're already rbac'd on the resource itself, so we're ok with those.